docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+      };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+});
+```
+
+To hook up your state link to Apollo Client, concatenate it to the other links
+in your Apollo Link chain. Your state link should be one of the last links in
+your chain. Then, pass your link chain to the Apollo Client constructor.
+
+```js
+const client = new ApolloClient({
+  cache: new InMemoryCache(),
+  link: ApolloLink.from([stateLink, new HttpLink()]),
+});
+```
+
+How do we differentiate a request for local data from a request that hits our
+server? In our query or mutation, we specify which fields are client-only with a
+`@client` directive. This tells our network stack to retrieve or update the data
+in the cache with our resolver map that we passed into our state link.
+
+```js
+const UPDATE_NETWORK_STATUS = gql`
+  mutation updateNetworkStatus($isConnected: Boolean) {
+    networkStatus(isConnected: $isConnected) @client {
+      isConnected
+    }
+  }
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+What if we want to access our network status data from another component? Since
+we don't know whether our `UPDATE_NETWORK_STATUS` mutation will fire before we
+try to access the data, we should guard against undefined values by providing an
+initial state for our query in our resolver map.
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
+  Query: {
+    networkStatus: () => null,
+  },
+});
+```
+
+How do we query the `networkStatus` from our component? Similar to mutations,
+just use a query & the `@client` directive! With Apollo Link, we can combine
+data sources, including your remote data, in one query.
+
+In this example, the `articles` field will either hit the cache or fetch from
+our GraphQL endpoint, depending on our fetch policy. Since `networkStatus` is
+marked with `@client`, we know that this is local data, so it will resolve from
+the cache.
+
+```js
+const GET_ARTICLES = gql`
+  query {
+    networkStatus @client {
+      isConnected
+    }
+    articles {
+      id
+      title
+    }
+  }
+`;
+```
+
+To retrieve the data in your component, bind your query to your component via
+your favorite Apollo view layer integration just like you normally would. Then,
+React Apollo will attach both your remote & local data to `props.data` while
+tracking both loading & error states. Once the query returns a result, your
+component will update reactively.
+
+```js
+const WrappedComponent = graphql(GET_ARTICLES, {
+  props: ({ data: { networkStatus, articles } }) => {
```

Comment:
Is this actually helpful? Do people actually commonly write `props` handlers instead of just taking the default? Maybe it would be more useful to demonstrate this working with no props option but with an ordinary component inside that displays the network status.

Reply:
At MLS, we always wrote custom props handlers, but I'm not sure what other people in the wild do. 🤷‍♀️  The main reason I did the handler was so people could see the loading & error states, but we could probably put that logic inside the component instead.